### PR TITLE
R4R:for some feature 0.10.2

### PR DESCRIPTION
**Pull request by wenwu-bianjie**
https://github.com/irisnet/explorer/issues/993
https://github.com/irisnet/explorer/issues/992
https://github.com/irisnet/explorer/issues/991
https://github.com/irisnet/explorer/issues/990
https://github.com/irisnet/explorer/issues/989

### CODE REVIEW

File: backend/rest/controller/tx.go [hunk 1]
```diff
@@ -63,6 +89,10 @@ func registerQueryTxListByType(r *mux.Router) error {
 		tx.SetTid(request.TraceId)
 		query := bson.M{}
 		total := GetString(request, "total")
+		txTyp := QueryParam(request, "txType")
```
Comment (kaifei-bianjie) [inline backend/rest/controller/tx.go]: rename `txTyp ` to `txType`;

File: backend/rest/controller/tx.go [hunk 2]
```diff
@@ -83,28 +113,50 @@ func registerQueryTxListByType(r *mux.Router) error {
...
 			}
-			return tx.QueryTxList(query, page, size, istotal)
-		case types.Gov:
-			query["type"] = bson.M{
-				"$in": types.GovernanceList,
+		}
+		if txTyp != "" {
+			query["type"] = txTyp
+		} else {
+			switch types.TxTypeFromString(txType) {
```
Comment (kaifei-bianjie) [inline backend/rest/controller/tx.go]: rename `txType ` to `txTypeGroup`

File: backend/types/const.go [hunk 3]
```diff
@@ -71,6 +71,7 @@ const (
 	UrlRegisterQueryTxsByAccount = "/txsByAddress/{address}/{page}/{size}"
 	UrlRegisterQueryTxsByDay     = "/txsByDay"
 	UrlRegisterQueryTx           = "/tx/{hash}"
+	UrlRegisterQueryTxType       = "/txType"
```
Comment (kaifei-bianjie) [inline backend/types/const.go]: rename `/txType` to `/tx_types/{type}`

File: backend/vo/tx.go [hunk 4]
```diff
@@ -125,19 +125,20 @@ func (t Tx) PrintHashFromToAmount() string {
...
+	Fee         utils.ActualFee   `json:"fee"`
+	Status      string            `json:"status"`
+	GasLimit    int64             `json:"gas_limit"`
+	GasUsed     int64             `json:"gas_used"`
+	GasWanted   int64             `json:"gas_wanted"`
+	GasPrice    float64           `json:"gas_price"`
+	Memo        string            `json:"memo"`
+	Log         string            `json:"log"`
+	Timestamp   time.Time         `json:"timestamp"`
+	Tags        map[string]string `json:"tags,omitempty"`
```
Comment (kaifei-bianjie) [inline backend/vo/tx.go]: delete `omitempty ` flag

File: backend/rest/controller/tx.go (hunk 1 above)
Comment (wenwu-bianjie) [reply backend/rest/controller/tx.go]: ok,modify

File: backend/rest/controller/tx.go (hunk 2 above)
Comment (wenwu-bianjie) [reply backend/rest/controller/tx.go]: have change

File: backend/types/const.go (hunk 3 above)
Comment (wenwu-bianjie) [reply backend/types/const.go]: changed

File: backend/vo/tx.go (hunk 4 above)
Comment (wenwu-bianjie) [reply backend/vo/tx.go]: modify

**Pull request merged**